File: src/leetcode/python/GasStation_134.py
class GasStation_134(object):
    def canCompleteCircuit(self, gas, cost):
        """
        :type gas: List[int]
        :type cost: List[int]
        :rtype: int
        """
        length = len(gas)
        complete = sum(gas[i] - cost[i] for i in range(length))
        if complete < 0: return -1

        start, sub = 0, 0
        for i in range(length):
            sub += gas[i] - cost[i]
            if sub < 0:
                sub = 0
                start = i + 1
        return start

File: src/leetcode/java/GasStation_134.java
package leetcode.java.medium;

public class GasStation_134 {
  public int canCompleteCircuit(int[] gas, int[] cost) {
    int start = -1;
    int complete = 0;
    for (int i = 0; i < gas.length; i++) complete += gas[i] - cost[i];
    if (complete < 0) return start;
    int sub = 0;
    start = 0;
    for (int i = 0; i < gas.length; i++) {
      sub += gas[i] - cost[i];
      if (sub < 0) {
        sub = 0;
        start = i + 1;
      }
    }
    return start;
  }
}
